tendme.lic

```diff
@@ -1,85 +1,81 @@
 =end
 
-custom_require.call(%w[common common-healing])
+custom_require.call(%w[common common-healing events])
 
-def bind_open_wounds?
-  DRC.bput('heal', '^You')
-  pause 1
-  raw = reget(15)
-  bleeders = raw
-             .dup
-             .reverse
-             .take_while { |item| item !~ /^Bleeding/ }
-             .keep_if { |item| item =~ /^\s+(?:\w+|\w+\s\w+)\s{7}.+/ }
-             .delete_if { |item| item =~ /\bskin\b/ }
-  leeches = raw
-            .dup
-            .map { |x| x.scan(/leech on your [^\,\.]+,|leech on your [^\,\.]+\./).to_a }
-            .flatten
-            .compact
-  mites = raw
-          .dup
-          .map { |x| x.scan(/mite on your [^\,\.]+,|mite on your [^\,\.]+\./).to_a }
-          .flatten
-          .compact
-  lodged = raw
-           .dup
-           .map { |x| x.scan(/lodged \w* into your [^\,\.]+/).to_a }
-           .flatten
-           .compact
+class TendMe
 
-  bleeders = bleeders.keep_if { |item| !item.include?('tended') && !item.include?('inside') && !item.include?('Area') }
-
-  lodged.each do |message|
-    message =~ /into your ([^\,\.]+)/
-    DRCH.bind_wound(Regexp.last_match(1))
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'train', regex: /train/i, optional: true, description: 'unwrap and rebind wounds for optimum learning' },
+        { name: 'monitor', regex: /monitor/i, optional: true, description: 'run in background to keep bleeders tended' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+    waitrt? while bind_open_wounds?
+    monitor_health(args.train) if args.monitor
   end
 
-  bleeders.each do |message|
-    message =~ /^\s+(\w+|\w+\s\w+)\s{7}/
-    DRCH.bind_wound(Regexp.last_match(1))
+  def bind_open_wounds?
+    health_data = DRCH.check_health
+    wounds = (
+      health_data['lodged'].values.flatten +
+      health_data['parasites'].values.flatten +
+      health_data['bleeders'].values.flatten
+    )
+    wounds
+      .reject { |wound| wound.internal? || wound.bleeding_rate =~ /tended|clotted/ }
+      .any? { |wound| DRCH.bind_wound(wound.body_part) }
   end
 
-  leeches.each do |message|
-    message =~ /on your ([^\,\.]+)|(\,|\.)/
-    DRCH.bind_wound(Regexp.last_match(1))
+  # Monitor for wounds to begin bleeding again then tend them.
+  def monitor_health(train)
+    Flags.add('tendme-bind-lodged', /You feel a slight pain from the (?<item>.*) lodged in your (?<body_part>.*)\./)
+    Flags.add('tendme-bind-wound', /The bandages binding your (?<body_part>.*) (become|come|soak)/)
+    Flags.add('tendme-rewrap-wound', /You feel like now might be a good time to change the bandages on your (?<body_part>.*)\./)
+
+    loop do
+      pause 0.5
+      body_part = nil
+      # One or more active bleeders, tend them.
+      if Flags['tendme-bind-wound']
+        body_part = get_body_part_from_flag('tendme-bind-wound')
+      end
+      # If training, you can rewrap wound for more experience.
+      if Flags['tendme-rewrap-wound'] && train
```

Comment:
Should this also check if field experience is full? I.e. even if the `train` arg was used, if it wouldn't give us an exp then don't rebind?

Reply:
Good idea, I'll add the EXP check

Reply:
Resolved via https://github.com/rpherbig/dr-scripts/pull/4916/commits/5a682a1143c3826083e3efab70434dea83266f66